optimize/util/optimize-commons/src/main/java/io/camunda/optimize/service/db/schema/OptimizeIndexNameService.java

```diff
@@ -114,20 +119,33 @@ public static String getOptimizeIndexOrTemplateNameForAliasAndVersion(
   public String getOptimizeIndexOrTemplateNameForAliasAndVersionWithPrefix(
       final String indexOrTemplateNameWithoutPrefix, final String version) {
     return getOptimizeIndexAliasForIndexNameAndPrefix(
-        getOptimizeIndexOrTemplateNameForAliasAndVersion(indexOrTemplateNameWithoutPrefix, version),
-        indexPrefix);
+        getOptimizeIndexOrTemplateNameForAliasAndVersion(
+            indexOrTemplateNameWithoutPrefix, version));
   }
 
-  public static String getOptimizeIndexAliasForIndexNameAndPrefix(
-      final String indexName, final String indexPrefix) {
-    String original = indexName;
-    if (!indexName.startsWith(indexPrefix)) {
-      original = String.join("-", indexPrefix, indexName);
+  public String getOptimizeIndexAliasForIndexNameAndPrefix(String indexName) {
+    indexName = String.join("-", COMPONENT_NAME, indexName);
+    if (!indexPrefix.isEmpty() && !indexName.startsWith(indexPrefix)) {
+      indexName = String.join("-", indexPrefix, indexName);
+    }
+    return indexName.toLowerCase(Locale.ENGLISH);
+  }
+
+  private static String makeCompleteIndexPrefix(final String indexPrefix) {
+    if (indexPrefix == null || indexPrefix.isEmpty()) {
+      return COMPONENT_NAME;
+    } else {
+      return String.format("%s-%s", indexPrefix, COMPONENT_NAME);
     }
-    return original.toLowerCase(Locale.ENGLISH);
   }
 
   public String getIndexPrefix() {
-    return indexPrefix;
+    return completeIndexPrefix;
```

Comment:
the complete prefix is then "{customPrefix}-optimize" so that when it's used for searching all optimize indices, it will not show indices from other apps as well (for example when indexPrefix is "")